lib/routes/emails.js

```diff
@@ -279,15 +282,23 @@ module.exports = (log, db, mailer, config, customs, push) => {
          * Below is a summary of the verify_code flow. This flow is used to verify emails, sign-in and
          * account codes.
          *
-         * 1) Check request against customs server, proceed if valid.
+         * 1) Check if signatures match, proceed if they do otherwise go to 2)
          *
-         * 2) If type=`secondary` then this is an email code and verify it
+         * 2) Check request against customs server, proceed if valid.
+         *
+         * 3) If type=`secondary` then this is an email code and verify it
          *    accordingly.
          *
-         * 3) Otherwise attempt to verify code as sign-in code then account code.
+         * 4) Otherwise attempt to verify code as sign-in code then account code.
          */
         return db.account(uid)
           .then((account) => {
+            const serverVerificationKey = config.serverVerification.key
+            if (isServerVerified(code, uid, serverVerificationKey, requestSignature)) {
+              log.info({op: 'account.recovery_email_verify.server_verification.success', code, uid})
+              return account
+            }
+            log.info({op: 'account.recovery_email_verify.server_verification.fail', code, uid})
```

Comment:
we only need to log the fail if `requestSignature` was in the payload and we failed validation